fili-core/src/main/java/com/yahoo/bard/webservice/web/ResponseFormatResolver.java

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+/**
+ * A functional interface which allows the logic of resolving response format customizable. It has one method that
+ * takes the format name string from URI and a ContainerRequestContext object from which the header Accept field is
+ * accessible.
+ */
+public interface ResponseFormatResolver {
```

Comment:
you might want to have this explicitly implement BiFunction<String, ContainerRequestContext, String>

Reply:
I'm new to SAM, but since it's an interface, implementing another interface is not allowed in Java, right? Or did I misunderstand what you mean by "implement BiFunction<..."

Reply:
Sorry, since it's an interface, extend, not implement.